bot/exts/valentines/valentine_zodiac.py

```diff
@@ -33,48 +25,55 @@ def __init__(self, bot: commands.Bot):
         self.zodiacs, self.zodiac_fact = self.load_comp_json()
 
     @staticmethod
-    def load_comp_json() -> dict:
+    def load_comp_json() -> Tuple[dict, dict]:
         """Load zodiac compatibility from static JSON resource."""
-        p1 = Path("bot/resources/valentines/zodiac_explanation.json")
-        p = Path("bot/resources/valentines/zodiac_compatibility.json")
-        with p1.open(encoding="utf8") as json_data:
+        explanation_file = Path("bot/resources/valentines/zodiac_explanation.json")
+        compatibility_file = Path("bot/resources/valentines/zodiac_compatibility.json")
+        with explanation_file.open(encoding="utf8") as json_data:
             zodiac_fact = load(json_data)
-        with p.open(encoding="utf8") as json_data:
+            for zodiac_data in zodiac_fact.values():
+                zodiac_data['start_at'] = date.fromisoformat(zodiac_data['start_at'])
+                zodiac_data['end_at'] = date.fromisoformat(zodiac_data['end_at'])
+        with compatibility_file.open(encoding="utf8") as json_data:
             zodiacs = load(json_data)
-            return zodiacs, zodiac_fact
+        return zodiacs, zodiac_fact
+
+    def error(self, zodiac: str) -> discord.Embed:
+        """Returns error embed."""
+        embed = discord.Embed()
+        embed.color = Colours.pink
+        err_comp = [f"`{i}` {zod_name}" for i, zod_name in enumerate(self.zodiac_fact.keys(), start=1)]
+        err = ("\n").join(err_comp)
+        error_msg = f"**{zodiac}** is not a valid zodiac sign, here is the list of valid zodiac signs."
+        embed.description = f"{error_msg}\n{err}"
+        log.info("Wrong Zodiac name provided")
+        return embed
 
     def zodiac_sign_verify(self, zodiac: str) -> discord.Embed:
```

Comment:
`zodiac_sign_verify` doesn't really express what it actually does, maybe this could be renamed to `zodiac_build_embed`?
```suggestion
    def zodiac_build_embed(self, zodiac: str) -> discord.Embed:
```